01-data-structures/04-hashes-part-1/hashclass.rb

```diff
@@ -2,26 +2,65 @@ class HashClass
 
   def initialize(size)
     @items = Array.new(size)
+    @size = size
   end
 
+  # set
   def []=(key, value)
-  end
+    index = index(key, @size)
+
+    # empty hash
+    if @items[index].nil?
+      @items[index] = HashItem.new(key, value)
+
+    # new key
+    elsif @items[index].key != key
+      resize()
+      self[key] = value
+
+      # same key, new value
+    elsif @items[index].key === key && @items[index].value != value
+      resize()
+      @items[index(key, @size)].value = value
 
+    # else - do nothing if same key same value (no else statement)
+    end
 
+    p @items
+    p @size
+  end
+
+  # get
   def [](key)
+    if @items[index(key, @size)].key === key
+      return @items[index(key, @size)].value
+    end
   end
 
   def resize
+    @size = 2 * @size
+    temp = Array.new(@size)
+
+    for entry in @items do
+      if entry
+        new_index = index(entry.key, @size)
+        temp[new_index] = entry
```

Comment:
This is not something you have to address due to the advanced nature of the case and lack of RSpec for it, but it is possible that `new_index` could be taken. Just because there were no collisions at, say, 5 capacity does not mean there are not at 8 capacity.